dbus/methods_test.go

```diff
@@ -1657,3 +1659,41 @@ func TestFreezer(t *testing.T) {
 
 	runStopUnit(t, conn, TrUnitProp{target, nil})
 }
+
+func TestListUnitProcesses(t *testing.T) {
+	target, err := util.GetRunningSlice() // This test should still pass even if the cmd is spawned in a child unit (i.e. session Scope) of the current Slice
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	conn := setupConn(t)
+	defer conn.Close()
+
+	cmd := exec.Command("/bin/sleep", "400")
+	err = cmd.Start()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cmd.Process.Kill()
+
+	pid := uint64(cmd.Process.Pid)
+
+	ctx := context.Background()
+	processes, err := conn.GetUnitProcessesContext(ctx, target)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	exists := false
+	for _, p := range processes {
+		if p.PID == pid && strings.HasPrefix(p.Command, "/bin/sleep") {
+			exists = true
+			t.Logf("Found %v\n", p)
+		}
+	}
+
+	if !exists {
+		t.Errorf("PID %d ('/bin/sleep 400') not found in current Slice unit's process list", pid)
```

Comment:
It looks like this test is failing on some CI runs. I suspect there is a race between spawning the child `sleep` and getting unit processes. It could be useful to retry the `GetUnitProcessesContext()` a few times until we get a non-empty set with `/bin/sleep`, or eventually timeout if we never get that.